gems/smithy-xml/lib/smithy-xml/builder.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+require 'base64'
+
+module Smithy
+  module Xml
+    # @api private
+    class Builder
+      include Smithy::Schema::Shapes
+
+      def initialize(options = {})
+        @indent = options.fetch(:indent, '')
+        @pad = options.fetch(:pad, '')
+      end
+
+      def build(shape, data, target = nil)
+        ref = shape.is_a?(ShapeRef) ? shape : ShapeRef.new(shape: shape)
+        target ||= []
+        @builder = DocBuilder.new(target: target, indent: @indent, pad: @pad)
+        structure(location_name(ref, ref.member_name), ref, data)
+        target.join
+      end
+
+      private
+
+      def shape(name, ref, value)
+        case ref.shape
```

Comment:
Do we need to cover Boolean shape here? Serialized as "true" or "false" according to smithy docs

Reply:
Yes, I believe it's handled by the default case `node(name, ref, value.to_s)`. The rest xml tests should exercise this for building.